src/plugin/memory-webview-main.ts

```diff
@@ -137,7 +140,8 @@ export class MemoryWebview implements vscode.CustomReadonlyEditorProvider {
         // and executes code based on the message that is received
         const webviewParticipant = this.setWebviewMessageListener(panel, initialMemory);
 
-        // initialize with configuration
+        // initialize web view content
+        this.setTitle(webviewParticipant, panel.title);
         this.setInitialSettings(webviewParticipant);
```

Comment:
I believe that both of these lines should be moved into the handler of the `readyType` notification handler in `setWebviewMessageListener`. Otherwise, there's no guarantee that the code that sets up the listeners to handle these requests will have run inside the webview - I suspect that that's why I'm seeing just `Memory` in the webview I created rather than the intended title. This also suggests that we should probably keep all of the message handling centralized in the `App` class, even if it's a bit inconvenient: that at least provides the guarantee that the messaging is synchronized.

Reply:
Excellent suggestion! Done! :+1: